tools/version-bump/lib/validate.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { valid, lt as versionLessThan, prerelease } from 'semver';
+import { readFile } from 'fs/promises';
+
+/**
+ * Internal dependencies
+ */
+import { Logger } from './logger';
+
+/**
+ * Get a plugin's current version.
+ *
+ * @param  plugin plugin to update.
+ */
+export const getCurrentVersion = async (
+	plugin: string
+): Promise< string | void > => {
+	try {
+		const composerJSON = JSON.parse(
+			await readFile( `plugins/${ plugin }/composer.json`, 'utf8' )
+		);
+		return composerJSON.version;
+	} catch ( e ) {
+		Logger.error( 'Unable to read current version.' );
+	}
+};
+
+/**
+ * When given a prerelease version, return just the version.
+ *
+ * @param {string} prereleaseVersion version with prerelease params
+ * @return {string} version
+ */
+export const stripPrereleaseParameters = (
+	prereleaseVersion: string
+): string => {
+	const prereleaseParameters = prerelease( prereleaseVersion );
+	if ( ! prereleaseParameters ) {
+		return prereleaseVersion;
+	}
+
+	const substr = prereleaseParameters.reduce( ( str, value, idx ) => {
+		if ( idx === 0 ) {
+			return `-${ value }`;
+		}
+		return `${ str }.${ value }`;
+	}, '' );
+
+	return prereleaseVersion.replace( substr.toString(), '' );
```

Comment:
I wonder if this version parsing could be done with `semver`. Its fine how it is, just food for thought.

Reply:
I scoured the docs for a method that does this, ie takes `7.1.0-beta.1` and returns `7.1.0` but I could not find anything! In the end, I opted to change this to use `parse` which breaks it down into an object and concatenate major, minor, and patch. Less logic seems better.